### PR TITLE
feat: add feature to share report

### PR DESCRIPTION
**Pull request by benw202**
## Description

This PR adds a feature for Admin, moderation, developer, media and CS to share their report with the rest of the team in the #mod-alerts channel

## Test Results

User interaction:
image

Mod alert:
image

Shared report:
image

## Discord Username

benw8484

### CODE REVIEW

File: src/commands/context/message/reportMessage.ts [hunk 1]
```diff
@@ -227,6 +223,74 @@ export default contextMenuCommand(data, async ({ interaction }) => {
...
+            await shareReportButtonInteraction.reply({
+                content: `Your report has been submitted and shared in ${modAlertsChannel}.`,
+                ephemeral: true,
+            });
+
+            await scamReportLogs.send({ content: `Reported message from ${targetMessage.author.toString()} has been shared in ${modAlertsChannel}.` });
+        } catch (error) {
+            Logger.error(error);
+            await interaction.followUp({
+                content: `You did not choose to share in time. Your report has been submitted without sharing in ${modAlertsChannel}.`,
```
Comment (Sleinmaster) [inline src/commands/context/message/reportMessage.ts]:
```suggestion
                content: `Your report has been submitted without sharing in ${modAlertsChannel}.`,
```

File: src/commands/context/message/reportMessage.ts [hunk 2]
```diff
@@ -227,6 +223,74 @@ export default contextMenuCommand(data, async ({ interaction }) => {
...
+            components: [
+                {
+                    type: 1,
+                    components: [
+                        {
+                            type: 2,
+                            style: 3,
+                            label: 'Yes',
+                            customId: 'shareReportYes',
+                        },
```
Comment (Sleinmaster) [inline src/commands/context/message/reportMessage.ts:249]: I would add a No Button as well that just skips the wait. Or even if it doesn't do anything, it feels better to be able to say no.

**Review by Sleinmaster — COMMENTED** [review]
Otherwise looks good to me.

File: src/commands/context/message/reportMessage.ts (hunk 2 above)
Comment (benw202) [reply src/commands/context/message/reportMessage.ts:249]: Yeah? I was going with less button presses if they didn't have to. I can add it in though

Comment (Sleinmaster) [reply src/commands/context/message/reportMessage.ts:249]: I would still leave the timeout in and have no as default, but still put the button, but that's my opinion. Doesn't have to be pressed, but also doesn't suggest that you have to say yes.

**Review by pdellaert — APPROVED** [review]

**Pull request merged**